rs/https_outcalls/consensus/src/pool_manager.rs

```diff
@@ -119,6 +120,8 @@ impl CanisterHttpPoolManagerImpl {
             .chain(
                 canister_http_pool
                     .get_unvalidated_shares()
+                    // Only check the unvalidated shares belonging to the requests that we can't validate.
+                    .filter(|share| share.content.id <= next_callback_id)
```

Comment:
Should it be `<`? `next_callback_id` sounds like it would hold `highest_callback_id_in_use + 1`
```suggestion
                    .filter(|share| share.content.id < next_callback_id)
```

Reply:
My thinking is: in case of doubt, just fetch one more because one share doesn't change the bounds complexity. But strictly less will not break anything, so I will change.

Reply:
Correct it doesn't change the bounds. But if the priority function gave us a share with `id == next_callback_id`, then that id is technically still in the future, and there should be no way that there is an active context for it yet. So if we went with `<=`, then we might mistakenly purge the share anyway.